src/sentry/seer/anomaly_detection/store_data.py

```diff
@@ -103,19 +116,52 @@ def update_rule_data(
         for k, v in updated_fields.items():
             setattr(alert_rule, k, v)
 
-        handle_send_historical_data_to_seer(alert_rule, project, "update")
+        for k, v in updated_query_fields.items():
+            if k == "dataset":
+                v = v.value
+            elif k == "time_window":
+                time_window = updated_query_fields.get("time_window")
+                v = (
+                    int(time_window.total_seconds())
+                    if time_window is not None
+                    else snuba_query.time_window
+                )
+            elif k == "event_types":
```

Comment:
`event_types` is stored in a different table and updated [here](https://github.com/getsentry/sentry/blob/master/src/sentry/snuba/subscriptions.py#L121-L126)